tests/by-util/test_cp.rs

```diff
@@ -3701,3 +3703,57 @@ fn test_cp_no_such() {
         .fails()
         .stderr_is("cp: 'no-such/' is not a directory\n");
 }
+
+#[cfg(all(unix, not(target_os = "macos")))]
+fn compare_xattrs<P: AsRef<Path>>(path1: P, path2: P) -> bool {
+    let get_sorted_xattrs = |path: P| {
+        xattr::list(path)
+            .map(|attrs| {
+                let mut attrs = attrs.collect::<Vec<_>>();
+                attrs.sort();
+                attrs
+            })
+            .unwrap_or_else(|_| Vec::new())
+    };
+
+    get_sorted_xattrs(path1) == get_sorted_xattrs(path2)
+}
+
+#[cfg(all(unix, not(any(target_os = "android", target_os = "macos"))))]
+#[test]
+fn test_acl_preserve() {
+    use std::process::Command;
+
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let path1 = "a";
+    let path2 = "b";
+    let file = "a/file";
+    let file_target = "b/file";
+    at.mkdir(path1);
+    at.mkdir(path2);
+    at.touch(file);
+
+    let path = at.plus_as_string(file);
+    // calling the command directly. xattr requires some dev packages to be installed
+    // and it adds a complex dependency just for a test
+    match Command::new("setfacl")
+        .args(["-m", "group::rwx", &file])
```

Comment:
There is a `setfacl: a/file: Not a directory` message when running the tests, so I guess it should be:
```suggestion
        .args(["-m", "group::rwx", &path1])
```